packages/react-composites/src/composites/MeetingComposite/adapter/AzureCommunicationMeetingAdapter.ts

```diff
@@ -454,6 +464,42 @@ export const createAzureCommunicationMeetingAdapter = async ({
   return new AzureCommunicationMeetingAdapter(callAdapter, chatAdapter);
 };
 
+/**
+ * Arguments for {@link createAzureCommunicationMeetingAdapterFromClient}
+ *
+ * @beta
+ */
+export type AzureCommunicationMeetingAdapterFromClientArgs = {
+  meetingLocator: CallAndChatLocator | TeamsMeetingLinkLocator;
+  callAgent: CallAgent;
+  callClient: StatefulCallClient;
+  chatClient: StatefulChatClient;
+  chatThreadClient: ChatThreadClient;
+};
+
+/**
+ * Create a {@link MeetingAdapter} using the provided {@link StatefulChatClient} and {@link StatefulCallClient}.
+ *
+ * Useful if you want to keep a reference to {@link StatefulChatClient} and {@link StatefulCallClient}.
+ * Consider using {@link createAzureCommunicationMeetingAdapter} for a simpler API.
+ *
+ * @beta
+ */
+export const createAzureCommunicationMeetingAdapterFromClients = async ({
+  callClient,
+  callAgent,
+  meetingLocator,
+  chatClient,
+  chatThreadClient
+}: AzureCommunicationMeetingAdapterFromClientArgs): Promise<MeetingAdapter> => {
+  const locator = isTeamsMeetingLinkLocator(meetingLocator) ? meetingLocator : meetingLocator.callLocator;
+  const createCallAdapterPromise = createAzureCommunicationCallAdapterFromClient(callClient, callAgent, locator);
```

Comment:
Do we have a preference to have these steps or can we just await when we are creating them and then skip having to do a Promise.all later?

Reply:
The `Promise.all` here allows them to complete asynchronously - so the second adapter doesn't have to wait on the first if the first has any IO work (e.g. network calls) that it takes up time waiting for to complete